tango/test_context.py

```diff
@@ -250,9 +316,6 @@ def connect(self):
         # Get server proxy
         self.server = DeviceProxy(self.get_server_access())
         self.server.ping()
-        # Get device proxy
-        self.device = DeviceProxy(self.get_device_access())
-        self.device.ping()
```

Comment:
Sounds good. I'm proceeding with the change.

Reply:
Actually, with the tests that I did, I observe that the ping to dserver device replies only after all the other devices are exported. Another note is that the ping is not "waiting". If there is no connection, the ping simply raises an exception. 

IMO, ping to all the others devices is "just in case". Do you know any example of what could go wrong so the other devices does not respond to ping and the dserver responds?

I imagine that in Sardana we will make a generic test context that will populate "sar_demo" device server with approx. 30 devices. Then in particular tests we will just use proxies to, let's say, a couple of them. I think that preparing all the 30 device proxies up-front is not optimal. What I see good is to maintain cache of the device proxies within the context so it will be filled gradually on calls to `get_device`. Or simple don't maintain this cache and leave it to the test developer to keep them and not call `get_device` every time it is necessary to do something with the device.

I'll wait with implementing these changes for your opinion.

Reply:
> Actually, with the tests that I did, I observe that the ping to dserver device replies only after all the other devices are exported. Another note is that the ping is not "waiting". If there is no connection, the ping simply raises an exception.

OK, good to know.  OK, I see, the ping will not wait.

> IMO, ping to all the others devices is "just in case". Do you know any example of what could go wrong so the other devices does not respond to ping and the dserver responds?

No, no examples.

> I imagine that in Sardana we will make a generic test context that will populate "sar_demo" device server with approx. 30 devices. Then in particular tests we will just use proxies to, let's say, a couple of them. I think that preparing all the 30 device proxies up-front is not optimal. What I see good is to maintain cache of the device proxies within the context so it will be filled gradually on calls to `get_device`. Or simple don't maintain this cache and leave it to the test developer to keep them and not call `get_device` every time it is necessary to do something with the device. 

That's an interesting use case.  With a little effort, we could get the best of both.  I'm thinking we could create a dict-like object that lazy loads the device proxies.  We return this to the user (or they access it via a `self.devices` property).  The first time a key is accessed, the `DeviceProxy` object is created.  What do you think?

Reply:
Done in 0dac6a9.

Finally, to make it simpler, I went for an option for directly keeping the dictionary within the context class scope. A dedicated class for this purpose would need to know the host, port, nodb to be able to compose the Tango URI. This is because we want to get the devices using just the family/domain/member names, as these are used in the `devices_info` argument.

Reply:
OK.